layouts/BlogCategoryLayout.tsx

```diff
@@ -8,45 +8,56 @@ import Link from '@/components/Link';
 import Pagination from '@/components/Pagination';
 import getBlogData from '@/next-data/blogData';
 
-const getCurrentCategory = (pathname: string) => {
+const getCategoryData = async (pathname: string) => {
   // We split the pathname to retrieve the blog category from it since the
   // URL is usually /blog/{category} the second path piece is usually the
   // category name, which usually year-YYYY
   const [, _pathname, category] = pathname.split('/');
-
   if (_pathname === 'blog' && category && category.length) {
-    return category;
+    const data = await getBlogData(category);
+    return {
+      ...data,
+      category,
+    };
   }
 
-  // if either the pathname does not match to a blog page
-  // which should not happen (as this hook should only be used in blog pages)
-  // or if there is no category in the URL we return the previous year as category name
-  // which is always the default category (for example, the blog index)
-  // see https://github.com/nodejs/nodejs.org/issues/6205 for a permanent solution
-  return `year-${new Date().getFullYear() - 1}`;
+  // If the pathname does not match to a blog page,
+  //   which should not happen (as this hook should only be used in blog pages),
+  // or, if there is no category in the URL,
+  //   which happens when we're on the blog overview page (index),
+  // then we return the most recent year with blog posts
+  let year = new Date().getFullYear();
```

Comment:
I think this logic is unnecessary, just make:

```tsx
const year = Math.max(...data.meta.pagination);

const data = await getBlogData(`year-${year}`);

return { ...data, category: `year-${year}` };
```

Reply:
Ah right, you're right.

Reply:
Can we add inline docs on the reasoning behind then and maybe some blank lines, it's just the standard we do here, we separate definitions from assertions.

```tsx
let data = await getBlogData(`year-${year}`);

if (!data.posts.length && data.meta.pagination.length) {
  year = Math.max(...data.meta.pagination);
  data = await getBlogData(`year-${year}`);
}

return { ...data, category: `year-${year}` };
```